forte/data/data_store.py

```diff
@@ -244,9 +254,26 @@ def _new_group(self, type_name: str, member_type: str) -> List:
 
         tid: int = self._new_tid()
         entry = [member_type, [], tid, type_name]
-        entry += len(self._type_attributes[type_name]) * [None]
+        try:
+            entry += len(self._type_attributes[type_name]) * [None]
+        except KeyError as e:
+            raise KeyError(
+                f"Group type with id {type_name} does not exist."
+            ) from e
         return entry
 
+    def _is_annotation(self, type_name: str) -> bool:
+        r"""This function takes a type_id and returns whether a type
+        is an annotation type or not.
+        Args:
+            type_name (str): The name of type in `self.__elements`.
+        Returns:
+            A boolean value whether this type_id belongs to an annotation
+            type or not.
+        """
+        entry_class = get_class(type_name)
+        return issubclass(entry_class, Annotation)
```

Comment:
https://github.com/asyml/forte/pull/676#discussion_r824927567. Or maybe we can fix this issue in [#628 ](https://github.com/asyml/forte/pull/628#discussion_r825503299)

Reply:
I will solve it in a new PR #687. This API will be kept.